base/file_name_utils.hpp

```diff
@@ -50,7 +55,6 @@ std::string JoinPath(std::string const & folder, Args &&... args)
 template <typename... Args>
 std::string JoinPath(std::string const & dir, std::string const & fileOrDir, Args &&... args)
 {
-  ASSERT(!dir.empty(), ("JoinPath dir is empty"));
   ASSERT(!fileOrDir.empty(), ("JoinPath fileOrDir is empty"));
   return impl::JoinPath(dir, fileOrDir, std::forward<Args>(args)...);
 }
```

Comment:
There's something I don't like.
How come that `arg1` can be empty and `arg2` can't?

Reply:
you are right, in this implementation there is no reason why `fileOrDir` can not be an empty string as well.
I deleted the second `ASSERT` as well.